HeavyIonsAnalysis/PhotonAnalysis/plugins/ggHiNtuplizer.cc

```diff
@@ -836,6 +858,34 @@ void ggHiNtuplizer::fillElectrons(const edm::Event& e, const edm::EventSetup& es
   edm::Handle<edm::View<reco::GsfElectron> > gsfElectronsHandle;
   e.getByToken(gsfElectronsCollection_, gsfElectronsHandle);
 
+  // Get the electron ID data from the event stream.
+  // Note: this implies that the VID ID modules have been run upstream.
+  // If you need more info, check with the EGM group.
+  edm::Handle<edm::ValueMap<bool> > veto_id_decisions;
+  edm::Handle<edm::ValueMap<bool> > loose_id_decisions;
+  edm::Handle<edm::ValueMap<bool> > medium_id_decisions;
+  edm::Handle<edm::ValueMap<bool> > tight_id_decisions; 
+  e.getByToken(eleVetoIdMapToken_ ,veto_id_decisions);
```

Comment:
I think this will crash for the PbPb samples since we don't run the ID there. Can you add a boolean guard so that getByToken is only run when doEleVID is true, or something like that?